crux-bench/src/crux/bench.clj

```diff
@@ -109,59 +107,61 @@
-    :crux.kafka/doc-topic "kafka-rocksdb-doc"
-    :crux.kafka/tx-topic "kafka-rocksdb-tx"
-    :crux.kv/db-dir (str (io/file data-dir "kv/rocksdb"))}
+   (fn [data-dir]
+     {:crux.node/topology '[crux.kafka/topology
+                            crux.metrics/with-cloudwatch
+                            crux.kv.rocksdb/kv-store]
+      :crux.kafka/bootstrap-servers "localhost:9092"
+      :crux.kafka/doc-topic "kafka-rocksdb-doc"
+      :crux.kafka/tx-topic "kafka-rocksdb-tx"
+      :crux.kv/db-dir (str (io/file data-dir "kv/rocksdb"))})
 
    #_"standalone-lmdb"
-   #_{:crux.node/topology '[crux.standalone/topology
-                            crux.metrics/with-cloudwatch
-                            crux.kv.lmdb/kv-store]
-      :crux.kv/db-dir (str (io/file data-dir "kv/lmdb"))
-      :crux.standalone/event-log-kv-store 'crux.kv.lmdb/kv
-      :crux.standalone/event-log-dir (str (io/file data-dir "eventlog/lmdb"))}
+   #_(fn [data-dir]
+       {:crux.node/topology '[crux.standalone/topology
+                              crux.metrics/with-cloudwatch
+                              crux.kv.lmdb/kv-store]
+        :crux.kv/db-dir (str (io/file data-dir "kv/lmdb"))
+        :crux.standalone/event-log-kv-store 'crux.kv.lmdb/kv
+        :crux.standalone/event-log-dir (str (io/file data-dir "eventlog/lmdb"))})
 
    #_"kafka-lmdb"
-   #_{:crux.node/topology '[crux.kafka/topology
-                            crux.metrics/with-cloudwatch
-                            crux.kv.lmdb/kv-store]
-      :crux.kafka/bootstrap-servers "localhost:9092"
-      :crux.kafka/doc-topic "kafka-lmdb-doc"
-      :crux.kafka/tx-topic "kafka-lmdb-tx"
-      :crux.kv/db-dir (str (io/file data-dir "kv/rocksdb"))}})
-
-(defn with-nodes* [f]
+   #_(fn [data-dir]
+       {:crux.node/topology '[crux.kafka/topology
+                              crux.metrics/with-cloudwatch
+                              crux.kv.lmdb/kv-store]
+        :crux.kafka/bootstrap-servers "localhost:9092"
+        :crux.kafka/doc-topic "kafka-lmdb-doc"
+        :crux.kafka/tx-topic "kafka-lmdb-tx"
+        :crux.kv/db-dir (str (io/file data-dir "kv/rocksdb"))})})
+
+(defn with-nodes* [nodes f]
   (f/with-tmp-dir "dev-storage" [data-dir]
     (with-open [emb (ek/start-embedded-kafka
                       {:crux.kafka.embedded/zookeeper-data-dir (str (io/file data-dir "zookeeper"))
                        :crux.kafka.embedded/kafka-log-dir (str (io/file data-dir "kafka-log"))
                        :crux.kafka.embedded/kafka-port 9092})]
       (vec
-        (for [[k v] (nodes data-dir)]
-          (with-open [node (api/start-node v)]
-            (with-dimensions {:crux-node-type k}
-              (log/infof "Running bench on %s node." k)
-              (post-to-slack (str "running on node: " k))
-              [k (f node)])))))))
-
-(defmacro with-nodes [[node-binding] & body]
-  `(with-nodes* (fn [~node-binding] ~@body)))
+       (for [[k ->node] nodes]
```

Comment:
A nitpick - and not necessarily one with your code: there has to be a better/clearer varname for the node's name than just 'k' (I realize it is the 'key' but not particularly descriptive or helpful)

Reply:
`node-type`?

Reply:
Yeah, I think that's far better.